compiler/passes/checkParsed.cpp

```diff
@@ -124,6 +130,25 @@ checkNamedArguments(CallExpr* call) {
   }
 }
 
+static const char* dotAstr;
+static VarSymbol* deinitStrLiteral;
+
+static void setupForCheckExplicitDeinitCalls() {
+  dotAstr = astr(".");
+  deinitStrLiteral = new_CStringSymbol("deinit");
+}
+
+static void checkExplicitDeinitCalls(CallExpr* call) {
+  if (call->id == breakOnResolveID) gdbShouldBreakHere(); //vass
+  if (UnresolvedSymExpr* target = toUnresolvedSymExpr(call->baseExpr))
+    if (target->unresolved == dotAstr)
+      if (SymExpr* arg2 = toSymExpr(call->get(2)))
+        if (arg2->symbol() == deinitStrLiteral)
+          // OK to invoke explicitly from internal modules
+          if (strcmp(call->parentSymbol->name, "chpl__delete"))
```

Comment:
The comment above and the actual code don't match. It says OK to invoke from internal modules; so I'd expect you to be checking if it's in an internal module. Instead you check if it's in chpl__delete. At least make the comment and the code consistent.